functests/sctp/sctp.go

```diff
@@ -0,0 +1,219 @@
+			}, 1*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodSucceeded))
+		})
+	})
+})
+
+func loadMC() *mcfgv1.MachineConfig {
+	decode := mcfgScheme.Codecs.UniversalDeserializer().Decode
+	mcoyaml, err := ioutil.ReadFile(mcYaml)
+	Expect(err).ToNot(HaveOccurred())
+
+	obj, _, err := decode([]byte(mcoyaml), nil, nil)
+	Expect(err).ToNot(HaveOccurred())
+	mc, ok := obj.(*mcfgv1.MachineConfig)
+	Expect(ok).To(Equal(true))
+	return mc
+}
+
+func checkForSctpReady(cs *client.ClientSet) {
+	nodes, err := cs.Nodes().List(metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker-sctp=",
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	args := []string{`set -x; x="$(checksctp 2>&1)"; echo "$x" ; if [ "$x" = "SCTP supported" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`}
+	for _, n := range nodes.Items {
+		job := jobForNode("checksctp", n.ObjectMeta.Labels[hostnameLabel], "checksctp", []string{"/bin/bash", "-c"}, args)
+		cs.Pods(testNamespace).Create(job)
+	}
+
+	Eventually(func() bool {
+		pods, err := cs.Pods(testNamespace).List(metav1.ListOptions{LabelSelector: "app=checksctp"})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+		for _, p := range pods.Items {
+			if p.Status.Phase != k8sv1.PodSucceeded {
+				return false
+			}
+		}
+		return true
+	}, 3*time.Minute, 10*time.Second).Should(Equal(true))
+}
+
+func createSctpService(cs *client.ClientSet) {
+	service := k8sv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "sctpservice",
+			Namespace: testNamespace,
+		},
+		Spec: k8sv1.ServiceSpec{
+			Selector: map[string]string{
+				"app": "sctpserver",
+			},
+			Ports: []k8sv1.ServicePort{
+				k8sv1.ServicePort{
+					Protocol: k8sv1.ProtocolSCTP,
+					Port:     30101,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.String,
+						StrVal: "sctpserver",
+					},
+				},
+			},
+		},
+	}
+	_, err := cs.Services(testNamespace).Create(&service)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func scptTestPod(name, node, app string, args []string) *k8sv1.Pod {
+	res := k8sv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: name,
+			Labels: map[string]string{
+				"app": app,
+			},
+			Namespace: testNamespace,
+		},
+		Spec: k8sv1.PodSpec{
+			RestartPolicy: k8sv1.RestartPolicyNever,
+			Containers: []k8sv1.Container{
+				{
+					Name:    name,
+					Image:   testerImage,
+					Command: []string{"/usr/bin/sctptest"},
+					Args:    args,
+				},
+			},
+			NodeSelector: map[string]string{
+				hostnameLabel: node,
+			},
+		},
+	}
+
+	return &res
+}
+
+func jobForNode(name, node, app string, cmd []string, args []string) *k8sv1.Pod {
+	job := k8sv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: name,
+			Labels: map[string]string{
+				"app": app,
+			},
+			Namespace: testNamespace,
+		},
+		Spec: k8sv1.PodSpec{
+			RestartPolicy: k8sv1.RestartPolicyNever,
+			Containers: []k8sv1.Container{
+				{
+					Name:    name,
+					Image:   "quay.io/wcaban/net-toolbox:latest",
```

Comment:
can we move this image into your or a team repo?  the user can change the image at any time and break our CI

Reply:
Agree, I'll use the same one we use for the tester (see my note above) in a different pr